hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/clustering/run/strategy/MultipleSparkJobExecutionStrategy.java

```diff
@@ -108,7 +108,7 @@ public HoodieWriteMetadata<HoodieData<WriteStatus>> performClustering(final Hood
     Stream<HoodieData<WriteStatus>> writeStatusesStream = FutureUtils.allOf(
             clusteringPlan.getInputGroups().stream()
                 .map(inputGroup -> {
-                  if (getWriteConfig().getBooleanOrDefault("hoodie.datasource.write.row.writer.enable", false)) {
+                  if (getWriteConfig().getBooleanOrDefault("hoodie.datasource.write.row.writer.enable", true)) {
```

Comment:
cc @nsivabalan . 

Good catch. It looks like we cannot use the ConfigProperty directly due to circular dependency. Can you comb the codebase to see if there are similar cases ?

Reply:
guess this was intentional. unless user explicitly enables this config, we don't want to enable it for clustering.

Reply:
lets also consider issues like https://github.com/apache/hudi/issues/8259 before we can make it default.